src/components/AvailabilityGrid.tsx

```diff
@@ -287,150 +302,222 @@ const Grid = ({
-            (time, index) =>
-              index !== timeArray.length && (
+  return (
+    <div className="flex flex-col">
+      <div onMouseUp={handleMouseUp} onMouseLeave={handleMouseLeaveGrid}>
+        {/* Column Header for weekdays corresponding to specific dates */}
+        <div>
+          {mode === 'specific' && (
+            <div
+              style={{
+                display: 'grid',
+                gridTemplateColumns: `repeat(${dimensions.width || 1}, 1fr)`,
+                marginLeft: '62px',
+              }}
+            >
+              {config?.map((dateString: string, index: number) => (
                 <div
                   key={index}
-                  className="flex items-start justify-start border-gray-300 pr-2 text-xs text-gray-600"
-                  style={{ height: '64px', lineHeight: '64px' }}
+                  className="flex flex-col items-center justify-center border-gray-300 text-xs text-gray-400"
                 >
-                  {time}
+                  <div>
+                    {new Date(dateString).toLocaleDateString('en-US', {
+                      weekday: 'short',
+                    })}
+                  </div>
                 </div>
-              ),
+              ))}
+            </div>
           )}
         </div>
 
-        <div>
-          {/* Grid Column header displaying days of the week */}
+        <div
+          className="grid-container"
+          style={{ display: 'grid', gridTemplateColumns: 'auto 1fr' }}
+        >
+          {/* Time row headers */}
           <div
-            className="grid-header"
+            className="grid-time-headers"
             style={{
               display: 'grid',
-              gridTemplateColumns: `repeat(${dimensions.width || 1}, 1fr)`, // Ensure at least one column
+              gridTemplateRows: `repeat(${dimensions.height}, 32px)`,
+              alignItems: 'flex-start',
             }}
           >
-            {(dates?.length as number) > 0 ? (
-              dates?.map((day, index) => (
-                <div
-                  key={index}
-                  className="flex items-start justify-center border-gray-300 text-[15px] text-gray-600"
-                  style={{ height: '2rem' }}
-                >
-                  {day}
-                </div>
-              ))
-            ) : (
-              // Placeholder for grid header (fixes alignment of times and grid rows when no dates are present)
-              <div
-                className="flex items-center justify-center border-gray-300 text-sm text-gray-600"
-                style={{ height: '2rem' }}
-              ></div>
+            {timeArray.map(
+              (time, index) =>
+                index !== timeArray.length && (
+                  <div
+                    key={index}
+                    className="flex items-start justify-start border-gray-300 pr-2 text-xs text-gray-600"
+                    style={{ height: '64px', lineHeight: '64px' }}
```

Comment:
any specific reason for this styling? just curious.

Reply:
I think the styling is the same as it was before.